tasks/Task.cpp

```diff
@@ -458,3 +410,113 @@ std::string Task::getStringOfSettings(DeviceSettings settings, SourceLevel sourc
             << getStringOfSettings(settings);
     return text.str();
 }
+
+void Task::enqueueRawPacket(iodrivers_base::RawPacket const &raw_packet, DeviceSettings const &pool_size)
+{
+    // Considering just the first and actual channel. Let a minimal free buffer for delivery an Instant Message.
+    if(raw_packet.data.size() > (pool_size.poolSize[0] - MAX_MSG_SIZE))
+    {
+        exception(HUGE_RAW_DATA_INPUT);
+        return;
+    }
+    if(queueRawPacket.size() > MAX_QUEUE_RAW_PACKET_SIZE)
+    {
+        exception(FULL_RAW_DATA_QUEUE);
+        return;
+    }
+    queueRawPacket.push(raw_packet);
+}
+
+void Task::enqueueSendIM(SendIM const &sendIM)
+{
+    // Check size of Message. It can't be bigger than MAX_MSG_SIZE, according device's manual.
+     if(sendIM.buffer.size() > MAX_MSG_SIZE)
+     {
+         exception(HUGE_INSTANT_MESSAGE);
+         return;
+     }
+     // Check queue size.
+     if(queueSendIM.size() > MAX_QUEUE_MSG_SIZE)
+     {
+         exception(FULL_IM_QUEUE);
+         return;
+     }
+     queueSendIM.push(sendIM);
+}
+
+bool Task::isSendIMAvbl(AcousticConnection const& acoustic_connection)
+{
+    // TODO define exactly what to do for each acoustic_connection status
+    if( acoustic_connection.status != ONLINE && acoustic_connection.status != INITIATION_ESTABLISH && acoustic_connection.status != INITIATION_LISTEN )
+        return false;
+    // Other Instant Message is been transmitted
+    if( driver->getIMDeliveryStatus() == PENDING)
+        return false;
+    // No Instant Message to transmit
+    if( queueSendIM.empty())
+        return false;
+    // Wait for ack of previous Instant Message
+    if( im_wait_ack)
+        return false;
+    // Check free buffer size
+    if( acoustic_connection.freeBuffer[0] < MAX_MSG_SIZE)
+        return false;
+    return true;
+}
+
+bool Task::isTransRawDataAvbl( AcousticConnection const& acoustic_connection)
+{
+    // TODO define exactly what to do for each acoustic_connection status
+    if( acoustic_connection.status != ONLINE && acoustic_connection.status != INITIATION_ESTABLISH && acoustic_connection.status != INITIATION_LISTEN )
+        return false;
+    // No Raw Data to transmit
+    if( queueRawPacket.empty())
+        return false;
+    // Check free buffer size
+    if( acoustic_connection.freeBuffer[0] < queueRawPacket.front().data.size())
+        return false;
+    return true;
+}
+
+void Task::sendIM(void)
```

Comment:
Was easy for me ... When reading the code I asked myself whether the methods were sending everything or only one message :P